x/emissions/module/abci.go

```diff
@@ -33,10 +33,11 @@ func EndBlocker(ctx context.Context, am AppModule) error {
 		wg.Add(1)
 		go func(topic types.Topic) {
 			defer wg.Done()
-			// Check the cadence of inferences
+			// Check the cadence of inferences, and just in case also check multiples of epoch lengths
+			// to avoid potential situations where the block is missed
```

Comment:
good call. I'm guessing this is the main reason why you moved `AddChurnReadyTopic()` here (out of `EmitRewards`, yeah?

Reply:
AddChurnReady has been moved here, because EmitRewards should not be a dependency to making requests (which was causing the process to never start making requests).

Then, the use of multiple of EpochLength (instead of `> 2*EpochLength` ) is more efficient than the previous set up.